rfc/reorder-waiting-executions.md

```diff
@@ -0,0 +1,116 @@
+
+
+## Timeline
+_The code for CLI command reorder implementation is almost ready for review. 
+Planning to implement this feature in two Phases._
+
+_Phase-1: completing CLI command implementation and make this feature available for customers to use and work on feedback if any._
+
+_Phase-2: Implementing GUI for reorder functionality by using the backend that is available from CLI reorder implementation in Phase-1._
+
+_Proposed timeline for Phase-1 would be 1-2 weeks as this feature is demoed for some Spinnaker users and got the feedback.
+Phase-2 would be 3-4 weeks as there are some existing issues in the waiting executions area of Spinnaker dashboard that needs to be fixed._
+
+## Design
+
+_The Re-ordering waiting executions of a pipeline can be done through Spin CLI command and Spin dashboard._
+
+_Introducing a new CLI command for reorder functionality of a waiting executions, the command will be used to Re-order waiting execution to UP/DOWN or Top/Bottom based on the current position of the provided execution._
+
+_Example Spin CLI Command_
+ >>spin pipeline execution reorder --execution-id 01GJQ601MC3SBACVNZB0XC5GHJ --reorder-action UP
+
+_New buttons will be added in Spinnaker dashboard(GUI) for each waiting execution to move the specific execution towards UP/Down or Top/Bottom based on the current position of the execution._
+
+_New Gate API endpoint will be created for external access_
+>>:8084/pipelines/<execution_id>/reorder?reorderAction=<UP/Down/Top/Bottom>
+
+_New Orca API endpoint will be created to access the same way_
+>>:8083/pipelines/<execution_id>/reorder?reorderAction=<UP/Down/Top/Bottom>
+
+_The main business logic of the feature lies in Spin Orca service, where the **CompoundExecutionOperator** will update the Redis repo using the existing class **RedisExecutionRepository** and update the pending executions to the user selected position in the Redis repo using the Class **RedisPendingExecutionService**._
+
+_A new handler class **ReorderWaitingExecutionHandler** will be created in orca-queue handler package, to handle the reorder of a waiting execution._
+
+**_Note_**
+
+_The Re-ordering will not be done for the executions which are not in Waiting state, an appropriate error message should be displayed._
+
+_In the Spinnaker dashboard(GUI), only the waiting executions should have buttons to reorder, buttons should be removed once the execution is started._
+
+
+### Dependencies
+
+_This proposal depends on existing services Spin CLI, Spin Deck, Spin Gate and Spin Orca._
+
+_No new dependencies are needed, existing systems will be sufficient to implement this feature_
+
+## Drawbacks
+
+_Currently there is no Re-order functionality of an execution_
+_Why should we **not** do this?_
+
+## Prior Art and Alternatives
```

Comment:
Have you looked at the `orca-qos` API? There may be a way to extend that to implement this functionality.